src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MemoryPoolViewBufferScope.cs

```diff
@@ -120,7 +120,7 @@ public void Dispose()
 
                 for (var i = 0; i < _leased.Count; i++)
                 {
-                    _viewBufferPool.Return(_leased[i]);
+                    _viewBufferPool.Return(_leased[i], clearArray: true);
```

Comment:
That's only called when something really strange happens and code that will never throw throws. It's basically there as a safeguard in case someone adds more code to that block.